x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/SourceDestValidator.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.dataframe.transforms;
+
+import org.elasticsearch.ElasticsearchStatusException;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.regex.Regex;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.xpack.core.dataframe.DataFrameMessages;
+import org.elasticsearch.xpack.core.dataframe.transforms.DataFrameTransformConfig;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+public final class SourceDestValidator {
+
+    /**
+     * Validates the DataFrameTransformConfiguration source and destination indices.
```

Comment:
nit: you say validate here but the method name is check

I wonder: there is more validation in `PutDataFrameTransformAction::validate` which also touches destIndex, maybe consolidate?

Reply:
@hendrikmuhs I was thinking around that, but all those validations on naming don't need to occur here. 

I can add them for completeness's sake, if this is a strongly desired. But the idea of these validations is that they are more complex than a simple "are the names valid" and require access to the cluster state.

Reply:
validation now happens in 2 places, this isn't obvious to someone who starts reading the code, so minimum some code comment would be nice. Consolidating all the logic in one place would be better but I am not sure if there are technical reasons why it's spread over 2 places. I am not asking for duplication.